### PR TITLE
Add support for webpacker

### LINKED ISSUE #3: Add webpacker support
Rails 5.1 includes webpacker by default. This requires config for yarn. Please add this to the example. Here's my working config.yml:

```yaml
version: 2
jobs:
  build:
    working_directory: ~/myapp
    docker:
      - image: circleci/ruby:2.3-node
        environment:
          PGHOST: 127.0.0.1
          PGUSER: myapp-test
          RAILS_ENV: test
      - image: circleci/postgres:9.5-alpine
        environment:
          POSTGRES_USER: myapp
          POSTGRES_DB: myapp-test
          POSTGRES_PASSWORD: ""
    steps:
      - checkout

      - type: cache-restore
        name: Restore bundle cache
        key: myapp-bundle-{{ checksum "Gemfile.lock" }}

      - type: cache-restore
        name: Restore yarn cache
        key: myapp-yarn-{{ checksum "yarn.lock" }}

      - run:
          name: Bundle Install
          command: bin/bundle install --path vendor/bundle

      - run:
          name: Yarn Install
          command: yarn install

      - type: cache-save
        name: Store bundle cache
        key: myapp-bundle-{{ checksum "Gemfile.lock" }}
        paths:
          - vendor/bundle

      - type: cache-save
        name: Store yarn cache
        key: myapp-yarn-{{ checksum "yarn.lock" }}
        paths:
          - ~/.yarn-cache

      - run:
          name: Rubocop
          command: bin/rubocop --rails

      - run:
          name: Wait for DB
          command: dockerize -wait tcp://localhost:5432 -timeout 1m

      - run:
          name: Database setup
          command: bin/rails db:schema:load --trace

      - run:
          name: Run tests
          command: bin/rails test
```

### PR DESCRIPTION
**Pull request by swrobel**
Fixes #3

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,29 +3,57 @@ jobs:
...
+      # Only necessary if app uses webpacker or yarn in some other way
+      - type: cache-save
+        name: Store yarn cache
+        key: rails-demo-yarn-{{ checksum "yarn.lock" }}
+        paths:
+          - ~/.yarn-cache
+
+      - run:
+          name: Wait for DB
+          command: dockerize -wait tcp://localhost:5432 -timeout 1m
```
Comment (zzak) [inline .circleci/config.yml]: Was this necessary?

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,29 +3,57 @@ jobs:
...
 
-      # Store bundle cache
       - type: cache-save
-        key: rails-demo-{{ checksum "Gemfile.lock" }}
+        name: Store bundle cache
+        key: rails-demo-bundle-{{ checksum "Gemfile.lock" }}
         paths:
           - vendor/bundle
 
-      # Database setup
```
Comment (zzak) [inline .circleci/config.yml]: Removed the comments :(

File: .circleci/config.yml [hunk 3]
```diff
@@ -3,29 +3,57 @@ jobs:
...
+      # Only necessary if app uses webpacker or yarn in some other way
+      - run:
+          name: Yarn Install
+          command: yarn install
 
-      # Store bundle cache
       - type: cache-save
-        key: rails-demo-{{ checksum "Gemfile.lock" }}
+        name: Store bundle cache
+        key: rails-demo-bundle-{{ checksum "Gemfile.lock" }}
```
Comment (zzak) [inline .circleci/config.yml]: Let's keep the save/restore steps around the install step, so users can easily copy/paste the parts they care about

File: .circleci/config.yml [hunk 4]
```diff
@@ -3,29 +3,57 @@ jobs:
...
         environment:
+          PGHOST: 127.0.0.1
+          PGUSER: circleci-demo-ruby
           RAILS_ENV: test
-      - image: circleci/postgres:9.4.12-alpine
+      - image: circleci/postgres:9.5-alpine
+        environment:
+          POSTGRES_USER: circleci-demo-ruby
+          POSTGRES_DB: rails_blog
+          POSTGRES_PASSWORD: ""
```
Comment (zzak) [inline .circleci/config.yml]: I didn't need to override these before, but have seen them in other projects, could you explain?

File: .circleci/config.yml [hunk 5]
```diff
@@ -3,29 +3,57 @@ jobs:
...
     working_directory: ~/circleci-demo-ruby-rails
     docker:
-      - image: circleci/ruby:2.4.1-node
+      - image: circleci/ruby:2.4-node
         environment:
+          PGHOST: 127.0.0.1
+          PGUSER: circleci-demo-ruby
           RAILS_ENV: test
-      - image: circleci/postgres:9.4.12-alpine
+      - image: circleci/postgres:9.5-alpine
```
Comment (zzak) [inline .circleci/config.yml]: Is 9.5 prevelant now? We should use whichever version is most common

File: .circleci/config.yml (hunk 1 above)
Comment (ryanwohara) [inline .circleci/config.yml]: It's safer to have it, especially with Alpine images.

**Comment by zzak** [comment]
@swrobel could you take a look at our comments here? I'm happy to help address them if you don't mind :)

**Comment by swrobel** [comment]
Go for it

File: .circleci/config.yml (hunk 1 above)
Comment (nplusp) [inline .circleci/config.yml]: Yup, it's really useful sometimes.

**Pull request merged**